sycl/include/CL/sycl/handler.hpp

```diff
@@ -796,13 +796,42 @@ class __SYCL_EXPORT handler {
     }
   }
 
+  template <typename T>
+  using remove_cv_ref_t =
+      typename std::remove_cv<detail::remove_reference_t<T>>::type;
+
+  template <typename U, typename T>
+  using is_same_type = std::is_same<remove_cv_ref_t<U>, remove_cv_ref_t<T>>;
+
+  template <typename T> struct ShouldEnableSetArg {
+    static constexpr bool value =
+        std::is_trivially_copyable<T>::value
+#if CL_SYCL_LANGUAGE_VERSION && CL_SYCL_LANGUAGE_VERSION <= 121
+            && std::is_standard_layout<T>::value
+#endif
+        || is_same_type<sampler, T>::value // Sampler
```

Comment:
Why did you remove `remove_cv_ref_t<T>` ?

Reply:
`remove_cv_ref_t` is used inside `is_same_type` for both types.